yaks/lib/yaks/reader/json_api.rb

```diff
@@ -12,12 +12,18 @@ def call(parsed_json, _env = {})
         else
           attributes = parsed_json['data'].dup
           links = attributes.delete('links') || {}
-          embedded   = convert_embedded(links, included)
+          type  = attributes.delete('type')
+          attributes.merge!(attributes.delete('attributes') || {})
+
+          association_links, resource_links = links.partition { |_k, v| v.is_a?(Hash) }
+          embedded   = convert_embedded(Hash[association_links], included)
+          links      = convert_links(Hash[resource_links])
+
           Resource.new(
-            type: Util.singularize(attributes.delete('type')[/\w+$/]),
+            type: Util.singularize(type[/\w+$/]),
```

Comment:
what's the use case for this regexp actually? (I realize it was already there but still curious)